File: src/main/java/it/contrader/automative/model/Auto.java
package it.contrader.automative.model;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;




@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor

@Entity
@Table(name = "auto")
public class Auto implements Serializable {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private int id;
	
	@Column
	private String marca;
	
	@Column
	private String modello;
	
	@Column
	private String targa;
	
	@Column
	private String numeroTelaio;
	
	@Column
	private int cilindrata;
	
	@Column
	private int potenza;
	
	@Column
	private String cambio;
	
	@Column
	private int numeroPorte;
	
	@Column
	private String alimentazione;
	
	@Column
	private int kmAttuali;
	
	@Column
	private int kmInizioNoleggio;
	
	@Column
	private Date scadenzaRevisione;
	
	@Column
	private Date scadenzaTagliando;
	
	@Column
	private Date scadenzaAssicurazione;
	
	@Column
	private Date scadenzaBollo;
	
	@Column
	private String tipologiaAuto;
	
	@Column
	private int daNoleggio;

//	@ManyToOne(fetch=FetchType.EAGER)
//	@JoinColumn(name = "utente_ID")
//	private Utente utente;
	
	@ManyToOne(fetch=FetchType.LAZY)
	@JoinColumn(name = "officina_ID")
	private Officina officina;
//	
//	@OneToOne(fetch=FetchType.EAGER)
//	@JoinColumn(name = "IdAuto")
//	private Auto auto;
	
}

File: FrontAutomotiveAngular5/src/app/models/Auto.ts
import { Officina } from "./Officina";

export class Auto {
    alimentazione:string;
    cilindrata:number;
    daNoleggio:number;
    id:number
    kmAttuali:number;
    kmInizioNoleggio:number;
    marca:string;
    modello:string;
    numeroPorte:number;
    numeroTelaio:string;
    officina:Officina;
    scadenzaAssicurazione:number
    scadenzaBollo:number
    scadenzaRevisione:number
    scadenzaTagliando:number
    targa:string
    tipologiaAuto:string


    constructor(alimentazione:string,cilindrata:number,daNoleggio:number,id:number,kmAttuali:number,
        kmInizioNoleggio:number,marca:string,modello:string,numeroPorte:number,numeroTelaio:string,
        officina:Officina,scadenzaAssicurazione:number,scadenzaBollo:number,scadenzaRevisione:number,
        scadenzaTagliando:number,targa:string,tipologiaAuto:string){
            this.alimentazione=alimentazione;
            this.cilindrata=cilindrata;
            this.daNoleggio=daNoleggio;
            this.id=id;
            this.kmAttuali=kmAttuali;
            this.kmInizioNoleggio=kmInizioNoleggio;
            this.marca=marca;
            this.modello=modello;
            this.numeroPorte=numeroPorte;
            this.numeroTelaio=numeroTelaio;
            this.officina=officina;
            this.scadenzaAssicurazione=scadenzaAssicurazione;
            this.scadenzaBollo=scadenzaBollo;
            this.scadenzaRevisione=scadenzaRevisione;
            this.scadenzaTagliando=scadenzaTagliando;
            this.targa=targa;
            this.tipologiaAuto=tipologiaAuto;
        }
}